supervisor/exceptions.py

```diff
@@ -1,9 +1,22 @@
 """Core Exceptions."""
 
 
+from typing import Callable, Optional
+
+
 class HassioError(Exception):
     """Root exception."""
 
+    def __init__(
+        self,
+        message: Optional[str] = None,
+        logger: Optional[Callable[..., None]] = None,
+    ) -> None:
+        """Raise & log."""
+        if logger:
+            logger(message)
```

Comment:
```suggestion
        logger: Optional[Callable[..., None]] = None,
        severity: Optional[str] = None,
    ) -> None:
        """Raise & log."""
        if logger or severity:
            (logger or _LOGGER.__getattribute__(severity))(message)
```

Can we add a severity argument for when a specific logger is not needed?

Reply:
We should not. A message should log in the context of who it was executed, also the severity would be in 70% just be a dump string "error" repeated 100 times in the code.